lib/sqlalchemy/orm/properties.py

```diff
@@ -652,17 +652,15 @@ def _init_column_for_annotation(
     ) -> None:
         sqltype = self.column.type
 
-        nullable = False
+        if is_fwd_ref(argument):
+            argument = de_stringify_annotation(cls, argument)
```

Comment:
I moved this up to here as I needed the forward refs from the tests to be resolved before trying to work out if they are optional.